lib/middleware/session.js

```diff
@@ -79,7 +79,14 @@ var ServletSession = exports.ServletSession = function(request) {
                 // to the attributes in the servlet session object.
                 data = new JavaAdapter(org.mozilla.javascript.NativeObject, {
                     put: function(name, start, value) {
-                        getSession().setAttribute(name, value);
+                    	if (typeof(value) === 'undefined' || value === null) {
+                    		getSession().removeAttribute(name);
+                    		if (Object.prototype[name]) {
+                                Object.prototype[name] = value;
```

Comment:
i strongly doubt it.

the new lines with `Object.prototype[name] = value` is dangerous (it unsets Object.prototype properties) and the old lines (reading from Object.prototype) were added by me: they don't work like that for functions and the commit has no clear goals https://github.com/ehoogerbeets/stick/commit/d797ca110dec6730b3c167445aa116b8462eb809#diff-fc3188fefb94c1c6ee0b3018183c93afR85